.github/workflows/tests.yaml

```diff
@@ -41,3 +42,33 @@ jobs:
         run: go build -v ./...
       - name: Run all tests
         run: go test -v ./...
+
+  shell-tests:
+    runs-on: macos-12
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version-file: ./go.mod
+      - name: Build devbox
+        run: go install ./cmd/devbox
+      - name: Install additional shells
+        env:
+          HOMEBREW_NO_INSTALL_CLEANUP: 1
+          HOMEBREW_NO_ENV_HINTS: 1
+          HOMEBREW_NO_EMOJI: 1
+          HOMEBREW_NO_ANALYTICS: 1
+        run: |
+          brew update
+          brew install dash zsh
+      - name: Install Nix
+        run: sh <(curl -L https://nixos.org/nix/install) --daemon
```

Comment:
Does nix get installed every single time the test runs? Can we cache this somehow

Reply:
I ask because if we install nix every single time, devbox shell would be quite slow given we have to install the nix store?

Reply:
Instead of installing nix, could we use a base image that has nix pre-installed? (for example, the same one we use in the dockerfile)

Reply:
Oh, nvm, I see you're purposefully trying to test macos.

Reply:
Installing Nix does slow things down a lot. With these tests the total CI time is about 5 minutes.

I want to look into caching the Nix and Homebrew packages with the cache action, but as a first step I wanted to get the tests working.